tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -66,11 +69,74 @@ static int64_t multiply_dims(llvm::ArrayRef<int64_t> dims, int64_t res = 1) {
   return res;
 }
 
-static int64_t count_dynamic_dims(llvm::ArrayRef<int64_t> dims) {
-  int64_t count = 0;
-  for (auto dim : dims)
-    if (ShapedType::isDynamic(dim)) ++count;
-  return count;
+static OpFoldResult multiply_dims(ArrayRef<OpFoldResult> dims,
+                                  OpBuilder& builder, Location loc,
+                                  OpFoldResult init = nullptr) {
```

Comment:
Given this just multiplies numbers together, just drop init and always use the first value. If you have a value you wish to scale by you can post apply it in the use.